pkg/storage/backend-storage/backend-storage.go

```diff
@@ -71,24 +74,127 @@ func IsBackendStorageNeededForVM(vm *corev1.VirtualMachine) bool {
 	return HasPersistentTPMDevice(&vm.Spec.Template.Spec)
 }
 
-func CreateIfNeeded(vmi *corev1.VirtualMachineInstance, clusterConfig *virtconfig.ClusterConfig, client kubecli.KubevirtClient) error {
+type BackendStorage struct {
+	client          kubecli.KubevirtClient
+	clusterConfig   *virtconfig.ClusterConfig
+	scStore         cache.Store
+	accessModeCache map[string]v1.PersistentVolumeAccessMode
+	cacheLock       sync.Mutex
+}
+
+func NewBackendStorage(client kubecli.KubevirtClient, clusterConfig *virtconfig.ClusterConfig, scStore cache.Store) *BackendStorage {
+	return &BackendStorage{
+		client:          client,
+		clusterConfig:   clusterConfig,
+		scStore:         scStore,
+		accessModeCache: map[string]v1.PersistentVolumeAccessMode{},
+	}
+}
+
+func (bs *BackendStorage) getStorageClass() (string, error) {
+	storageClass := bs.clusterConfig.GetVMStateStorageClass()
+	if storageClass != "" {
+		return storageClass, nil
+	}
+
+	for _, obj := range bs.scStore.List() {
+		sc := obj.(*storagev1.StorageClass)
+		if sc.Annotations["storageclass.kubernetes.io/is-default-class"] == "true" {
+			return sc.Name, nil
+		}
+	}
+
+	return "", fmt.Errorf("no default storage class found")
+}
+
+func (bs *BackendStorage) getMode(storageClass string, mode v1.PersistentVolumeMode) v1.PersistentVolumeAccessMode {
```

Comment:
confusion between volumemode and volumeaccessmode
```suggestion
func (bs *BackendStorage) getAccessMode(storageClass string, mode v1.PersistentVolumeMode) v1.PersistentVolumeAccessMode {
```